source/internal/numeric.d.ts

```diff
@@ -19,6 +19,25 @@ NumberAbsolute<NegativeInfinity>
 */
 export type NumberAbsolute<N extends number> = `${N}` extends `-${infer StringPositiveN}` ? StringToNumber<StringPositiveN> : N;
 
+/**
+Return a type if it is a number or a number string
+
+@example
+```
+type A = IsNumberLike<'1'>;
+//=> true
+
+type B = IsNumberLike<1>;
+//=> true
+
+type C = IsNumberLike<'a'>;
+//=> false
+*/
+export type IsNumberLike<N> =
```

Comment:
Does this support floating-point as a string? If it does, it should be documented, if not, it should be named `IsNumberOrIntegerString`.

It could also use a couple of tests.